File: contest/ABC308/C.jl
function main()
    N = parse(Int, readline())

    ans = Dict()
    input_data = readlines(stdin)
    for i in 1:N
        A, B = parse.(Int, split(input_data[i]))
        p = A // (A + B)  # 有理数として成功率を表現
        if !(p in keys(ans))
            ans[p] = [string(i)]
        else
            push!(ans[p], string(i))
        end
    end

    ans_s = sort(collect(keys(ans)), rev=true)
    ret = [ans[key] for key in ans_s]

    return join(join.(ret, " "), " ")
end

println(main())

File: contest/ABC308/C.py
"""
TLE
"""


import sys
from fractions import Fraction

def main():
    N = int(input())

    ans = {}
    input_data = sys.stdin.readlines()
    for i in range(N):
        A, B = map(int, input_data[i].split())
        p = Fraction(A, A + B)  # 有理数として成功率を表現
        if p not in ans:
            ans[p] = [str(i + 1)]
        else:
            ans[p].append(str(i + 1))

    ans_s = sorted(ans, reverse=True)
    ret = [ans[key] for key in ans_s]

    return ' '.join([' '.join(values) for values in ret])

if __name__ == '__main__':
    print(main())
